plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -21,33 +45,288 @@ public static GlobalKafkaConfiguration get() {
+        return getCredential(sslKeystoreCredentialsId).getUsername();
+    }
+
+    public String getSSLKeystorePassword() {
+        return getCredential(sslKeystoreCredentialsId).getPassword().getPlainText();
+    }
+
+    public String getSSLKeyPassword() {
+        return getCredential(sslKeyCredentialsId).getPassword().getPlainText();
+    }
+
+    public FormValidation doCheckBrokerURL(@QueryParameter("brokerURL") final String brokerURL) {
+        if (StringUtils.isBlank(brokerURL)) {
+            return FormValidation.error(Messages.GlobalKafkaConfiguration_KafkaConnectionURLWarning());
         }
         return FormValidation.ok();
     }
 
-    public FormValidation doCheckZookeeperURL(@QueryParameter String zookeeperURL) {
+    public FormValidation doCheckZookeeperURL(@QueryParameter("zookeeperURL") String zookeeperURL) {
         if (StringUtils.isBlank(zookeeperURL)) {
-            return FormValidation.warning(Messages.GlobalKafkaConfiguration_ZookeeperURLWarning());
+            return FormValidation.error(Messages.GlobalKafkaConfiguration_ZookeeperURLWarning());
         }
         return FormValidation.ok();
     }
 
+    public FormValidation doTestZookeeperConnection(@QueryParameter("zookeeperURL") final String zookeeperURL)
+            throws IOException, ServletException {
+        try {
+            String[] hostport = zookeeperURL.split(":");
+            String host = hostport[0];
+            int port = Integer.parseInt(hostport[1]);
+            testConnection(host, port);
+            return FormValidation.ok("Success");
+        } catch (Exception e) {
+            return FormValidation.error("Connection error : " + e.getMessage());
+        }
+    }
+
+    public FormValidation doTestBrokerConnection(@QueryParameter("brokerURL") final String brokerURL)
+            throws IOException, ServletException {
+        try {
+            String[] hostport = brokerURL.split(":");
+            String host = hostport[0];
+            int port = Integer.parseInt(hostport[1]);
+            testConnection(host, port);
+            return FormValidation.ok("Success");
+        } catch (Exception e) {
+            return FormValidation.error("Connection error : " + e.getMessage());
+        }
+    }
+
     @Override
-    public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
-        this.connectionURL = json.getString("connectionURL");
-        this.zookeeperURL = json.getString("zookeeperURL");
+    public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
+        this.brokerURL = formData.getString("brokerURL");
+        this.zookeeperURL = formData.getString("zookeeperURL");
+        this.enableSSL = Boolean.valueOf(formData.getString("enableSSL"));
         save();
-        return true;
+        return super.configure(req, formData);
+    }
+
+    private void testConnection(String host, int port) throws IOException {
+        new Socket(host, port);
```

Comment:
You do not try to connect the socket here. I doubt it checks much in the current state

Reply:
@oleg-nenashev this will create a socket to the host + port, if host and port are not available, an error message will be thrown. Is there anything I am missing here?